apps/sharedtest/test/unit/l10n/l10n_test.js

```diff
@@ -155,6 +155,23 @@ suite('L10n', function() {
       _translateFragment(elem);
       assert.equal(elem.getAttribute('aria-moz-hint'), 'moz-hint via ARIA');
     });
+
+    test('for a document fragment', function() {
+      var fragment = document.createDocumentFragment();
+      elem.dataset.l10nId = 'textcontent-test';
+      fragment.appendChild(elem);
+      _translateFragment(fragment);
+      assert.equal(elem.textContent, 'this is text content');
```

Comment:
Hah, you are right! We'll have to solve it when we update the API on our side. :)